src/Services/ForgotPassword.php

```diff
@@ -0,0 +1,125 @@
+<?php
+namespace App\Services;
+
+use App\Entity\User;
+use App\Services\Mailer;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+
+/**
+ * ForgotPassword class is for validate the userId to check whether the
+ * userId is coorect or not, generate and  send the otp to the registered
+ * email id of the user, verify the otp .
+ *
+ * @author rajdip <rajdip.roy@innoraft.com>
+ */
+class ForgotPassword
+{
+  /**
+   * This is a object of EntityManagerInterface class
+   * It is to manage persistance and retriveal Entity object from Database.
+   *
+   *   @var object
+   */
+  public $em;
+
+  /**
+   * This is a object of EntityRepository class
+   * It is to fetch data from user table of database.
+   *
+   *   @var object
+   */
+  public $userRepo;
+
+  /**
+   * This is a object of Mailer class, this helps to send mail.
+   *
+   *   @var object
+   */
+  public $mailer;
+
+  /**
+   * This constructor initializes object of ForgotPassword Class also provides
+   * access to EntityManagerInterface object .
+   *
+   * @param object $em
+   * It is to manage persistance and retriveal Entity object from Database.
+   *
+   * @return void
+   * Constructor returns nothing .
+   */
+  public function __construct(EntityManagerInterface $em)
+  {
+    session_start();
+    $this->em = $em;
+    $this->mailer = new Mailer();
+    $this->userRepo = $this->em->getRepository(User::class);
```

Comment:
Instead of using $em and $userRepo like this, you can send the $em and $userRepo in the object initialisation or send $userRepo object directly to the function as you have mentioned it in controller.